ext-src/packages/poetry/PoetryUtils.ts

```diff
@@ -26,12 +26,28 @@ export class PoetryUtils {
       const toml = readFileSync(join(application.workspaceFolder, "poetry.lock"));
       const poetry: Poetry = parse(toml.toString());
 
+      let isPyprojectTomlExists = false;
+      let productionDependencies: Set<string> = new Set<string>();
+
+      if (!includeDev) {
+        try {
+          let pyprojectToml = readFileSync(join(application.workspaceFolder, "pyproject.toml"));
+          isPyprojectTomlExists = true;
+          productionDependencies = this.resolveProductionDependencies(pyprojectToml, poetry);
+        } catch (ignored) {}
+      }
+
       if (poetry.package && poetry.package.length > 0) {
         let res: Array<PyPIPackage> = new Array();
 
         poetry.package.forEach(pkg => {
-          if (!includeDev && pkg.category == 'dev') {
-            return;
+          if (!includeDev) {
+            if (pkg.category == 'dev') {
```

Comment:
Will not work as intended due to: https://github.com/python-poetry/poetry/pull/7637/files